File: src/dsa/recursion/factorial/factorial.c
#include <stdio.h>
int factorial(int number);
int sum(int number);
int main()
{
    int num;
    printf("Enter number to find factorial : ");
    scanf("%d", &num);
    printf("Factorial of %d is : %d\n", num, factorial(num));
    printf("Sum of %d is : %d\n", num, sum(num));
    return 0;
}

int factorial(int number)
{
    if (number == 0)
    {
        return 1;
    }
    return number * factorial(number - 1);
}

int sum(int number)
{
    if (number == 1)
    {
        return 1;
    }
    return number + sum(number - 1);
}

File: src/dsa/recursion/factorial/Factorial.java
package dsa.recursion.factorial;

import java.util.Scanner;

public class Factorial {
    public static void main(String args[]) {

        Scanner in = new Scanner(System.in);
        System.out.println("Enter number to find factorial :");
        int num = Integer.parseInt(in.nextLine());
        System.out.println("Factorial of " + num + " is : " + factorial(num));
        Factorial fact = new Factorial();
        System.out.println("Sum of " + num + " is : " + fact.sum(num));
    }

    public static int factorial(int num) {
        if (num == 0) {
            return 1;
        }
        return num * factorial(num - 1);
    }

    public int sum(int num) {
        if (num == 1) {
            return 1;
        }
        return num + sum(num - 1);
    }
}